02-express-mongo-crud/mongoose-reference-populate/27-28/README.md

```diff
@@ -0,0 +1,359 @@
+
+1. Line 3: When we use `find` without a callback, then `populate`, like here, we can put a callback inside an `.exec()` method call. Technically we have made a query with `find`, but only executed it when we call `.exec()`.
+
+1. Lines 4-12: If we have any errors, we will log them.  Otherwise, we can display the entire Game Document **including** the populated consoles array.
+
+1. Line 13 demonstrates that we are able to access both data from the original Game Document we found **and** the referenced Console Document we summoned.
+
+<details>
+  <summary>What is the actual output from the above `findOne()` method call with populate?</summary>
+
+  ```js
+  { _id: 55e4eb857d6157f4d41a2981,
+    name: 'The Legend of Zelda: Ocarina of Time',
+    developer: 'Nintendo',
+    __v: 1,
+    consoles:
+     [ { _id: 55e4eb857d6157f4d41a2980,
+         name: 'Nintendo 64',
+         manufacturer: 'Nintendo',
+         released: Sun Sep 29 1996 00:00:00 GMT-0700 (PDT),
+         __v: 0 }
+     ]
+  }
+
+  I love The Legend of Zelda: Ocarina of Time on the Nintendo 64
+  ```
+</details>
+
+Now, instead of seeing **only** the `ObjectId` that pointed us to the `Console` document, we can see the **entire** `Console` document. Notice that the `Console` document's `_id` is exactly the same as the `ObjectId` that was previously referencing it. They are one in the same!
+
+
+## Integrate with Express
+
+### RESTful Route Design
+
+Remember to always make "RESTful" routes. RESTful routes are the most popular modern convention for designing resource paths for nested data. Here is an example of an application that has routes for `Store` and `Item` models:
+
+**RESTful Routing**
+
+| **HTTP Verb** | **Path** | **Description** |
+|---|---|---|
+| GET | /store | Get all stores |
+| POST | /store | Create a store |
+| GET | /store/:id | Get a store |
+| DELETE | /store/:id | Delete a store |
+| GET | /store/:store_id/items | Get all items from a store |
+| POST | /store/:store_id/items | Create an item for a store |
+| GET | /store/:store_id/items/:item_id | Get an item from a store |
+| DELETE | /store/:store_id/items/:item_id | Delete an item from a store |
+
+*In routes, avoid nesting resources deeper than shown above.*
+
+
+### Special Routes for Referenced Data
+
+Most of your routes for creating each piece of data will be the same, since games and consoles can exist independently of one another. However, when getting information about a game, remember to pull console data in with populate. Here are a few examples:
+
+```js
+// send all information for a single game
+app.get('/api/games/:gameId/', function (req, res) {
+  Game.findOne({ id: req.params.gameId })
+    .populate('consoles')
+    .exec(function(err, game) {
+      if (err) { res.status(500).send(err); }
+      console.log('found and populated game: ', game);
+      res.json(game);
+    });
+});
+```
+
+```js
+// send all information for all games
+app.get('/api/games/', function (req, res) {
```

Comment:
Maybe add some discussion that for an index route you might NOT send all this data.